libraries/AP_TemperatureSensor/AP_TemperatureSensor_DroneCAN.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AP_TemperatureSensor_config.h"
+
+#if AP_TEMPERATURE_SENSOR_DRONECAN_ENABLED
+
+#include "AP_TemperatureSensor_DroneCAN.h"
+#include <AP_BoardConfig/AP_BoardConfig.h>
+#include <AP_Math/AP_Math.h>
+
+AP_TemperatureSensor_DroneCAN* AP_TemperatureSensor_DroneCAN::_drivers[];
+uint8_t AP_TemperatureSensor_DroneCAN::_driver_instance;
+HAL_Semaphore AP_TemperatureSensor_DroneCAN::_driver_sem;
+
+extern const AP_HAL::HAL &hal;
+
+const AP_Param::GroupInfo AP_TemperatureSensor_DroneCAN::var_info[] = {
+
+    // @Param: MSG_ID
+    // @DisplayName: Temperature sensor DroneCAN message ID
+    // @Description: Sets the message device ID this backend listens for
+    // @Range: 0 65535
+    AP_GROUPINFO("MSG_ID", 1, AP_TemperatureSensor_DroneCAN, _ID, 0),
```

Comment:
Do we want to use -1 as an indicator to accept any like we do for Battery? I'm happy either way.